### PR TITLE
layout: make `padding` and `border` use `Au` in `pbm`

### PR DESCRIPTION
**Pull request by atbrakhi**
`padding` and `border` in `pbm` should use app units instead of length

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

### CODE REVIEW

File: components/layout_2020/flexbox/layout.rs [hunk 1]
```diff
@@ -540,8 +543,8 @@ impl<'a> FlexItem<'a> {
             content_box_size,
             content_min_size,
             content_max_size,
-            padding: padding.map(|t| (*t).into()),
-            border: border.map(|t| (*t).into()),
+            padding: padding.map(|t| (*t)),
+            border: border.map(|t| (*t)),
```
Comment (Loirooriol) [inline components/layout_2020/flexbox/layout.rs:547]:
```suggestion
            padding,
            border,
```

File: components/layout_2020/flow/inline.rs [hunk 2]
```diff
@@ -619,13 +619,15 @@ impl<'a, 'b> InlineFormattingContextState<'a, 'b> {
...
+            self.current_line.inline_position += (inline_box_state.pbm.padding.inline_start +
                 inline_box_state.pbm.border.inline_start +
                 inline_box_state
                     .pbm
                     .margin
                     .inline_start
-                    .auto_is(Length::zero);
+                    .auto_is(Length::zero)
+                    .into())
+            .into();
```
Comment (Loirooriol) [inline components/layout_2020/flow/inline.rs:630]: Converting a Length back and forth seems a bit weird. Consider
```rs
            self.current_line.inline_position += Length::from(
                inline_box_state.pbm.padding.inline_start +
                    inline_box_state.pbm.border.inline_start,
            ) + inline_box_state
                .pbm
                .margin
                .inline_start
                .auto_is(Length::zero);
```

File: components/layout_2020/flow/inline.rs [hunk 3]
```diff
@@ -659,8 +661,13 @@ impl<'a, 'b> InlineFormattingContextState<'a, 'b> {
...
                 inline_box_state.pbm.border.inline_end +
-                inline_box_state.pbm.margin.inline_end.auto_is(Length::zero);
-            self.current_line_segment.inline_size += pbm_end;
+                inline_box_state
+                    .pbm
+                    .margin
+                    .inline_end
+                    .auto_is(Length::zero)
+                    .into();
+            self.current_line_segment.inline_size += pbm_end.into();
```
Comment (Loirooriol) [inline components/layout_2020/flow/inline.rs:670]: Ditto:

```rs
            let pbm_end = Length::from(
                inline_box_state.pbm.padding.inline_end + inline_box_state.pbm.border.inline_end,
            ) + inline_box_state.pbm.margin.inline_end.auto_is(Length::zero);
            self.current_line_segment.inline_size += pbm_end;
```

File: components/layout_2020/flow/mod.rs [hunk 4]
```diff
@@ -785,7 +781,10 @@ fn layout_in_flow_non_replaced_block_level_same_formatting_context(
         // the block direction. In that case, the ceiling for floats is effectively raised
         // as long as no floats in the overflowing content lowered it.
         sequential_layout_state.advance_block_position(
-            (block_size - content_block_size + pbm.padding.block_end + pbm.border.block_end).into(),
+            (block_size - content_block_size +
+                pbm.padding.block_end.into() +
+                pbm.border.block_end.into())
+            .into(),
```
Comment (Loirooriol) [inline components/layout_2020/flow/mod.rs:787]:
```suggestion
            Au::from(block_size - content_block_size) +
                pbm.padding.block_end +
                pbm.border.block_end,
```

File: components/layout_2020/flow/mod.rs [hunk 5]
```diff
@@ -968,7 +970,7 @@ impl NonReplacedFormattingContext {
                     containing_block,
                     &collapsed_margin_block_start,
                     &pbm,
-                    &content_size + &pbm.padding_border_sums,
+                    (&content_size.clone().into() + &pbm.padding_border_sums).into(),
```
Comment (Loirooriol) [inline components/layout_2020/flow/mod.rs:973]:
```suggestion
                    &content_size.clone() + &pbm.padding_border_sums.clone().into(),
```

File: components/layout_2020/flow/mod.rs [hunk 6]
```diff
@@ -1034,7 +1037,7 @@ impl NonReplacedFormattingContext {
                 // size of auto. Try to fit it into our precalculated placement among the
                 // floats. If it fits, then we can stop trying layout candidates.
                 if placement.try_to_expand_for_auto_block_size(
-                    (content_size.block + pbm.padding_border_sums.block).into(),
+                    (content_size.block + pbm.padding_border_sums.block.into()).into(),
```
Comment (Loirooriol) [inline components/layout_2020/flow/mod.rs:1040]:
```suggestion
                    Au::from(content_size.block) + pbm.padding_border_sums.block,
```

File: components/layout_2020/flow/mod.rs [hunk 7]
```diff
@@ -1086,30 +1089,32 @@ impl NonReplacedFormattingContext {
         // Margins can never collapse into independent formatting contexts.
         sequential_layout_state.collapse_margins();
         sequential_layout_state.advance_block_position(
-            (pbm.padding_border_sums.block +
-                content_size.block +
-                clearance.unwrap_or_else(Length::zero))
-            .into(),
+            pbm.padding_border_sums.block +
+                content_size.block.into() +
+                clearance.unwrap_or_else(Length::zero).into(),
```
Comment (Loirooriol) [inline components/layout_2020/flow/mod.rs:1094]: Either is fine, but a bit shorter like this:

```suggestion
            pbm.padding_border_sums.block +
                (content_size.block + clearance.unwrap_or_else(Length::zero)).into(),
```

**Review by Loirooriol — APPROVED** [review]
It would be good if you can avoid regressing the test, but it's fine if it's too hard

File: components/layout_2020/flow/inline.rs (hunk 2 above)
Comment (atbrakhi) [reply components/layout_2020/flow/inline.rs]: Thanks, updated

**Pull request merged**